docs/architecture.md

```diff
@@ -0,0 +1,58 @@
+# Doctrine Extensions Architecture
+
+## Targeted Compatibility
+
+- PHP >= 7.2
+- Doctrine ORM 2.6+, 3.0 (no ODM / MongoDB - ORM only)
+- Doctrine Annotations Driver only
+
+## File Structure
+
+Doctrine Extensions consists of a single [monorepo](https://gomonorepo.org), that is
+automatically split into multiple read-only repositories for individual installation.
+
+Inside [`src`](../src) are three primary directories:
+
+1. [`Common`](../src/Common): common files required for any/all extensions to function.
+   Foundational files to support all extensions, hooks into Doctrine's functionality, etc.
+   
+2. [`Extension`](../src/Extension): the individual extensions, e.g. [Sluggable](../src/Extension/Sluggable)
+
+3. [`Tests`](../src/Tests): Common test utilities
+
+## Metadata
+
+At the source of any Doctrine project is
+[metadata](https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/reference/metadata-drivers.html#metadata-drivers).
+
+Doctrine Extensions uses a custom Annotation driver to parse entities for their configuration.
+The resulting [metadata class](../src/Common/Metadata/ExtendedClassMetadata.php) holds
+the information needed for each extension to perform.
+
+> :rotating_light: Doctrine Extensions **only supports Annotations** at this time.
```

Comment:
Not really a review, but I would not even consider YAML at this point, Doctrine 3 will drop it anyway.

XML might be interesting... Maybe a chain driver, loading the mapping from xml, and using annotations for the extensions?

Reply:
Good to know about Doctrine 3. I haven't done much research for what's going to change there. I'll be looking into it more as we develop this.